.github/workflows/codecov.yaml

```
@@ -0,0 +1,55 @@
+name: Coverage
+
+on:
+  - pull_request
```

```suggestion
  - main
  - pull_request
```
run coverage on main as well